src/utils/gitlab_updater.rs

```diff
@@ -0,0 +1,435 @@
+use crate::error::Result;
+use crate::{db::Pool, Config};
+use chrono::{DateTime, Utc};
+use log::{debug, info, trace, warn};
+use once_cell::sync::Lazy;
+use postgres::Client;
+use regex::Regex;
+use reqwest::{
+    blocking::Client as HttpClient,
+    header::{HeaderMap, HeaderValue, ACCEPT, AUTHORIZATION, USER_AGENT},
+};
+use serde::Deserialize;
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+
+const APP_USER_AGENT: &str = concat!(
+    env!("CARGO_PKG_NAME"),
+    " ",
+    include_str!(concat!(env!("OUT_DIR"), "/git_version"))
+);
+
+pub fn graphql_update(ids: &[&str]) -> String {
+    format!(
+        "\
+query {{
+    projects(ids: {:?}) {{
```

Comment:
We should be using GraphQL variables instead of generating the query dynamically.

Reply:
True, I had some issues with it but I think I found a way. In any case, this query doesn't work, only `graphql_single` is working for the moment.